SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialDebugger.h"
+
+#include "Debug/DebugDrawService.h"
+#include "Engine/Engine.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GameFramework/Pawn.h"
+#include "GameFramework/PlayerController.h"
+#include "GameFramework/PlayerState.h"
+#include "Interop/SpatialReceiver.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Kismet/GameplayStatics.h"
+#include "Schema/AuthorityIntent.h"
+
+using namespace SpatialGDK;
+
+DEFINE_LOG_CATEGORY(LogSpatialDebugger);
+
+ASpatialDebugger::ASpatialDebugger(const FObjectInitializer& ObjectInitializer)
+	: Super(ObjectInitializer)
+{
+	PrimaryActorTick.bCanEverTick = true;
+	PrimaryActorTick.bStartWithTickEnabled = true;
+
+	bAlwaysRelevant = true;
+	bNetLoadOnClient = false;
+	bReplicates = true;
+
+	NetUpdateFrequency = 1.f;
+
+	NetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+
+	// For GDK design reasons, this is the approach chosen to get a pointer
+	// on the net driver to the client ASpatialDebugger.  Various alternatives
+	// were considered and this is the best of a bad bunch.
+	if (NetDriver &&
+		NetDriver->IsServer() == false)
```

Comment:
Style: NetDriver != nullptr 
and does this need to be on two lines?

Reply:
Josh: I'd prefer to have NetDriver->IsServer()

Reply:
will fix